taskcat/cfn/stack.py

```diff
@@ -0,0 +1,348 @@
+from uuid import UUID, uuid4
+from taskcat.client_factory import ClientFactory
+from datetime import datetime
+from typing import List
+import re
+from taskcat.cfn.template import Template
+import json
+import os
+from pathlib import Path
+import random
+import string
+
+GENERIC_ERROR_PATTERNS = [
+    r'(The following resource\(s\) failed to create: )',
+    r'(^Resource creation cancelled$)'
+]
+
+
+class StackStatus:
+    COMPLETE = ['CREATE_COMPLETE', 'UPDATE_COMPLETE', 'DELETE_COMPLETE']
+    IN_PROGRESS = ['CREATE_IN_PROGRESS', 'DELETE_IN_PROGRESS', 'UPDATE_IN_PROGRESS',
```

Comment:
can you clarify why this would be useful ?

Intent behind these groupings is mostly to be able to tell if your stack is in motion, or in a static state